ArduCopter/GCS_Mavlink.cpp

```diff
@@ -974,6 +974,36 @@ MAV_RESULT GCS_MAVLINK_Copter::handle_command_long_packet(const mavlink_command_
         GCS_MAVLINK_Copter::convert_COMMAND_LONG_to_COMMAND_INT(packet, packet_int);
         return handle_command_pause_continue(packet_int);
     }
+
+    case MAV_CMD_DO_CHANGE_ALTITUDE: {
+        // param1 : Altitude in meters
+        // param2 : Frame of new altitude MAV_FRAME   
+        uint8_t frame = packet.param2;
```

Comment:
move uint8_t frame = packet.param2; just before the if (frame != MAV_FRAME_GLOBAL_RELATIVE_ALT) {
if we aren't in guided , no need to create it.
and make it const.